corehq/apps/reports/standard/deployments.py

```diff
@@ -225,15 +226,58 @@ def user_locations(self, ancestors, location_types):
             for location_type in location_types
         ]
 
+    def get_bulk_ancestors(self, location_ids):
+        """
+        Returns the grouped ancestors for the location ids passed in the
+        dictionary of following pattern
+        {location_id_1: [self, parent, parent_of_parent,.,.,.,],
+        location_id_2: [self, parent, parent_of_parent,.,.,.,],
+
+        }
+        :param domain: domain for which locations is to be pulled out
+        :param location_ids: locations ids whose ancestors needs to be find
+        :param kwargs: extra parameters
+        :return: dict
+        """
+        where = Q(domain=self.domain, location_id__in=location_ids)
+        location_ancestors = SQLLocation.objects.get_ancestors(where)
+        location_by_id = {location.location_id: location for location in location_ancestors}
+        location_by_pk = {location.id: location for location in location_ancestors}
+
+        grouped_location = {}
+        for location_id in location_ids:
+
+            location_parents = []
+            current_location = location_by_id[location_id].id if location_id in location_by_id else None
+
+            while current_location is not None:
+                location_parents.append(location_by_pk[current_location])
+                current_location = location_by_pk[current_location].parent_id
+
+            grouped_location[location_id] = location_parents
+
+        return grouped_location
+
+    def include_location_data(self):
+
+        return (
+            toggles.LOCATION_COLUMNS_APP_STATUS_REPORT.enabled(self.request.couch_user.username,
+                                                               toggles.NAMESPACE_USER)
+            and
+            toggles.LOCATION_COLUMNS_APP_STATUS_REPORT.enabled(self.domain,
+                                                               toggles.NAMESPACE_DOMAIN)
```

Comment:
This will only work if the feature flag is enabled for the user AND the domain.  What's that about?  I'd use `enabled_for_request`, which will check for user OR domain, which is the expected behavior with feature flags.

Reply:
hah - beat me to it in the next commit

Reply:
good coincidental example of what we were discussing in last book club. :P